okhttp/src/main/java/okhttp3/internal/http/BridgeInterceptor.kt

```diff
@@ -0,0 +1,120 @@
+ * request. Then it proceeds to call the network. Finally it builds a user response from the network
+ * response.
+ */
+class BridgeInterceptor(private val cookieJar: CookieJar) : Interceptor {
+
+  @Throws(IOException::class)
+  override fun intercept(chain: Interceptor.Chain): Response {
+    val userRequest = chain.request()
+    val requestBuilder = userRequest.newBuilder()
+
+    val body = userRequest.body()
+    body?.let {
+      val contentType = body.contentType()
+      contentType?.let {
+        requestBuilder.header("Content-Type", contentType.toString())
+      }
+
+      val contentLength = body.contentLength()
+      if (contentLength != -1L) {
+        requestBuilder.header("Content-Length", contentLength.toString())
+        requestBuilder.removeHeader("Transfer-Encoding")
+      } else {
+        requestBuilder.header("Transfer-Encoding", "chunked")
+        requestBuilder.removeHeader("Content-Length")
+      }
+    }
+
+    if (userRequest.header("Host") == null) {
+      requestBuilder.header("Host", hostHeader(userRequest.url(), false))
+    }
+
+    if (userRequest.header("Connection") == null) {
+      requestBuilder.header("Connection", "Keep-Alive")
+    }
+
+    // If we add an "Accept-Encoding: gzip" header field we're responsible for also decompressing
+    // the transfer stream.
+    var transparentGzip = false
+    if (userRequest.header("Accept-Encoding") == null && userRequest.header("Range") == null) {
+      transparentGzip = true
+      requestBuilder.header("Accept-Encoding", "gzip")
+    }
+
+    val cookies = cookieJar.loadForRequest(userRequest.url())
+    if (!cookies.isEmpty()) {
+      requestBuilder.header("Cookie", cookieHeader(cookies))
+    }
+
+    if (userRequest.header("User-Agent") == null) {
+      requestBuilder.header("User-Agent", Version.userAgent())
+    }
+
+    val networkResponse = chain.proceed(requestBuilder.build())
+
+    HttpHeaders.receiveHeaders(cookieJar, userRequest.url(), networkResponse.headers())
+
+    val responseBuilder = networkResponse.newBuilder()
+        .request(userRequest)
+
+    if (transparentGzip
+        && "gzip".equals(networkResponse.header("Content-Encoding"), ignoreCase = true)
+        && HttpHeaders.hasBody(networkResponse)) {
+
+      networkResponse.body()?.let {
+        val responseBody = GzipSource(it.source())
+        val strippedHeaders = networkResponse.headers().newBuilder()
+            .removeAll("Content-Encoding")
+            .removeAll("Content-Length")
+            .build()
+        responseBuilder.headers(strippedHeaders)
+        val contentType = networkResponse.header("Content-Type")
+        responseBuilder.body(RealResponseBody(contentType, -1L, responseBody.buffer()))
+      }
+
+    }
+
+    return responseBuilder.build()
+  }
+
+  /** Returns a 'Cookie' HTTP request header with all cookies, like `a=b; c=d`.  */
+  private fun cookieHeader(cookies: List<Cookie>): String {
+    val cookieHeader = StringBuilder()
```

Comment:
Use `buildString` instead, can actually refactor this into a single-expression function:
```kotlin
private fun cookieHeader(cookies: List<Cookie>) = buildString {
  ...
}
```

Reply:
Good Idea!